src/components/forms/SignUpForm/SignUpForm.js

```diff
@@ -1,126 +1,153 @@
           <Col>
-            <Form.Label>Email</Form.Label>
-            <Form.Control required name="email" onChange={this.handleChange} />
+            <Form.Group>
+              <Form.Label>Email</Form.Label>
+              <Form.Control
+                id="email"
+                required
+                type="email"
+                onChange={this.handleDataChange}
+              />
+            </Form.Group>
           </Col>
           <Col>
-            <Form.Label>Contact Number</Form.Label>
-            <Form.Control
-              required
-              name="contactNumber"
-              onChange={this.handleChange}
-            />
+            <Form.Group>
+              <Form.Label>Contact Number</Form.Label>
+              <Form.Control
+                id="telephone"
+                required
+                type="tel"
+                pattern="[0-9]{9,11}"
+                title="Must be a valid mobile phone number"
+                onChange={this.handleDataChange}
+              />
+            </Form.Group>
           </Col>
         </Form.Row>
         <Form.Row className="mb-4">
           <Col>
-            <Form.Label>Password</Form.Label>
-            <Form.Control
-              required
-              id="password"
-              name="password"
-              type="password"
-              onChange={this.handleChange}
-            />
+            <Form.Group>
+              <Form.Label>Password</Form.Label>
+              <Form.Control
+                required
+                id="password"
+                name="password"
+                type="password"
+                onChange={this.handleDataChange}
+              />
+            </Form.Group>
           </Col>
           <Col>
-            <Form.Label>Confirm Password</Form.Label>
-            <Form.Control
-              required
-              id="confirm-password"
-              name="password_confirmation"
-              type="password"
-              onChange={this.handleChange}
-            />
+            <Form.Group>
+              <Form.Label>Confirm Password</Form.Label>
+              <Form.Control
+                required
+                id="confirmedPassword"
+                ref={this.confirmedPasswordRef}
+                name="password"
+                type="password"
+                onChange={this.handleConfirmPasswordChange}
+              />
+            </Form.Group>
           </Col>
         </Form.Row>
         <div className="text-center" style={{ margin: 'auto' }}>
-          <Button
-            type="submit"
-            className="btn-more-info"
-            onClick={this.handleSubmit}
-            block
-          >
+          <Button variant="primary" type="submit" className="btn-confirm" block>
             Sign me up!
           </Button>
         </div>
       </Form>
     );
   }
 }
+function mapStateToProps() {
+  return {};
+}
 
-export default SignUpForm;
+export default connect(mapStateToProps)(SignUpForm);
```

Comment:
You can actually just do `connect()(SignUpForm)` without a `mapStateToProps` iirc

Reply:
Yeah looking at it, does seem a bit silly